metaflow/metadata/metadata.py

```diff
@@ -406,10 +406,75 @@ def get_object(cls, obj_type, sub_type, filters, attempt, *args):
             pre_filter, attempt_int
         )
 
+    @classmethod
+    def mutate_user_tags_for_run(
+        cls, flow_id, run_id, tags_to_remove=None, tags_to_add=None
+    ):
+        """
+        Mutate the set of user tags for a run.
+
+        Removals logically get applied after additions.  Operations occur as a batch atomically.
+        Parameters
+        ----------
+        flow_id : str
+            Flow id, that the run belongs to.
+        run_id: str
+            Run id, together with flow_id, that identifies the specific Run whose tags to mutate
+
+        Return
+        ------
+        Run tags after mutation operations
+        """
+        # perform common validation, across all provider implementations
+        if tags_to_remove is None:
+            tags_to_remove = []
+        if tags_to_add is None:
+            tags_to_add = []
+        if not tags_to_add and not tags_to_remove:
+            raise MetaflowTaggingError("Must add or remove at least one tag")
+        if isinstance(tags_to_add, str):
```

Comment:
We do check things here so maybe we can do something to keep both modes?

Reply:
We will do that per https://outerboundsco.slack.com/archives/C023Y9JH36W/p1653349445888399